day3/slice_time_image.py

```diff
@@ -46,11 +56,37 @@ def slice_time_image(img, slice_times, TR):
     st_img : Nifti1Image
         A new copy of the input image with slice-time interpolation applied
     """
+
     # Get the image data as an array;
+    data = img.get_data()       # The current function structure leads to many loadings of the image file, making the process slower and increasing memory demands
```

Comment:
You should generally keep lines to 80 characters length, to make lines easier to read in a variety of text-editors and web-interfaces, like this one.

In fact, you don't need to do ``img.get_data()`` below to find the number of slices, you can just do ``img = nib.load(fname); n_slices = img.shape[2]``.   This doesn't load the data from disk - see http://nipy.org/nibabel/nibabel_images.html#the-image-data-array